akka-http/src/main/scala/akka/http/javadsl/server/Directives.scala

```diff
@@ -26,4 +28,96 @@ object Directives extends AllDirectives {
 
   @varargs override def getFromBrowseableDirectories(directories: String*): Route =
     super.getFromBrowseableDirectories(directories: _*)
+
+  /**
+   * Partially applying functions for the Java World!
```

Comment:
A bit too silly comment :)
We should explain what this does so people can understand why they'd use it.
Perhaps an in-line example too?

Since some Java developers may feel this is a very new concept.